[Ingest pipelines] Polish pipeline debugging workflow

alisonelizabeth opened the pull request:
This PR contains a number of general usability and design improvements.

I tried to break the work up logically by commit (although I didn’t do the best job toward the end 😅). To test, please follow the same instructions in https://github.com/elastic/kibana/pull/74964 and ensure there are no regressions. 

**Notable changes:**
- More robust logic to determine the previous processor input (“Data in” accordion on the output tab of the processor flyout)
- Better handling of re-running simulation 
    - From the initial flow ("Add documents" button):
        - User can click the “Run the pipeline” button OR click the “Output” tab to rerun simulation.
        - Clicking "Refresh output" will refresh the output AND the verbose output (to refresh processor results)
    - From the processor flyout:
        - Simulation will run when clicking the “Update” button OR the “Output” tab.
        - If configuration changes are made, but the user cancels the submission, simulation will rerun with the original processors.
          - Note: I hit some oddities keeping the form in sync when going back and forth between the configuration and output tabs, so if you could test this area that would be appreciated!
- Created a container component for the test_pipeline_flyout
- Created two components for editing and adding a processor. This was previously one with conditional logic (formerly `manage_processor_form`). I felt this was getting overly complex, and there was enough divergence that it made sense to create two separate components.
- Various UX improvements
    - New icons to represent statuses 
    - New layout for processor output tab
    - New design for the test actions section

**What’s missing:**
- Better UX for empty prompt. Michael provided a design for this, but it will require additional refactoring. I’m putting it on hold for now and will address separately. 
- There is still not a way to cancel/stop simulation (without refreshing page). Design pending.

### Screenshots
<img width="1262" alt="Screen Shot 2020-09-01 at 8 54 06 PM" src="https://user-images.githubusercontent.com/5226211/91991923-862bf780-ed01-11ea-87c4-0db2d2110c2a.png">
<img width="367" alt="Screen Shot 2020-09-01 at 8 54 14 PM" src="https://user-images.githubusercontent.com/5226211/91992113-c0959480-ed01-11ea-8ada-fdd6510089ab.png">
<img width="785" alt="Screen Shot 2020-09-01 at 8 54 27 PM" src="https://user-images.githubusercontent.com/5226211/91992048-abb90100-ed01-11ea-9e0a-4f6226934bce.png">
<img width="346" alt="Screen Shot 2020-09-01 at 9 44 18 AM" src="https://user-images.githubusercontent.com/5226211/91991973-9512aa00-ed01-11ea-889b-e92e04f91e44.png">
<img width="755" alt="Screen Shot 2020-09-02 at 9 43 15 AM" src="https://user-images.githubusercontent.com/5226211/91991976-9512aa00-ed01-11ea-8a77-736041cb5f89.png">
<img width="752" alt="Screen Shot 2020-09-02 at 9 43 25 AM" src="https://user-images.githubusercontent.com/5226211/91991981-95ab4080-ed01-11ea-9116-456f993493ac.png">
<img width="751" alt="Screen Shot 2020-09-02 at 9 43 36 AM" src="https://user-images.githubusercontent.com/5226211/91991982-9643d700-ed01-11ea-86bd-c94356d2226c.png">
<img width="724" alt="Screen Shot 2020-09-02 at 9 43 46 AM" src="https://user-images.githubusercontent.com/5226211/91991986-96dc6d80-ed01-11ea-92a3-ae02a9435b69.png">

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70748/)
* Commit: 74ba900994fa8ca1190028212b183fdd9e488cfc
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70748/execution/node/370/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70748/","number":"70748","commit":"74ba900994fa8ca1190028212b183fdd9e488cfc"}]}
PIPELINE-->

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71136/)
* Commit: 65be40a3f04ef4348ba01d7015cb9da49838ea84
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71136/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Execute xpack-kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71136/execution/node/656/log)
* [Execute xpack-kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71136/execution/node/764/log)

---

### [Test Failures](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71136/testReport)

<details><summary>X-Pack SAML API Integration Tests.x-pack/test/saml_api_integration/apis/security/saml_login·ts.apis SAML security SAML authentication finishing handshake should succeed if both SAML response and handshake cookie are provided</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71136/testReport//junit/X-Pack%20SAML%20API%20Integration%20Tests/x-pack_test_saml_api_integration_apis_security_saml_login·ts/apis_SAML_security_SAML_authentication_finishing_handshake_should_succeed_if_both_SAML_response_and_handshake_cookie_are_provided)


#### Standard Out
```
Failed Tests Reporter:
  - Test has failed 1 times on tracked branches: https://dryrun

[00:00:00]       │
[00:00:00]         │ info [o.e.c.m.MetadataCreateIndexService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1598662513453591583] [.apm-custom-link] creating index, cause [api], templates [], shards [1]/[1]
[00:00:00]         │ info [o.e.c.r.a.AllocationService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1598662513453591583] updating number_of_replicas to [0] for indices [.apm-custom-link]
[00:00:00]         └-: apis SAML
[00:00:00]           └-> "before all" hook
[00:00:00]           └-: security
[00:00:00]             └-> "before all" hook
[00:00:00]             └-: SAML authentication
[00:00:00]               └-> "before all" hook
[00:00:00]               └-> should reject API requests if client is not authenticated
[00:00:00]                 └-> "before each" hook: global before each
[00:00:00]                 │ info [o.e.c.m.MetadataMappingService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1598662513453591583] [.kibana_1/RpdNL8b1R6m5Xx-wMyEgzg] update_mapping [_doc]
[00:00:00]                 └- ✓ pass  (95ms) "apis SAML security SAML authentication should reject API requests if client is not authenticated"
[00:00:00]               │ info [o.e.c.r.a.AllocationService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1598662513453591583] current.health="GREEN" message="Cluster health status changed from [YELLOW] to [GREEN] (reason: [shards started [[.apm-custom-link][0]]])." previous.health="YELLOW" reason="shards started [[.apm-custom-link][0]]"
[00:00:00]               └-> does not prevent basic login
[00:00:00]                 └-> "before each" hook: global before each
[00:00:00]                 │ proc [kibana]   log   [02:48:54.021] [info][plugins][routes][security] Logging in with provider "basic" (basic)
[00:00:00]                 │ info [o.e.x.i.a.TransportPutLifecycleAction] [kibana-ci-immutable-ubuntu-16-tests-xxl-1598662513453591583] adding index lifecycle policy [kibana-event-log-policy]
[00:00:00]                 │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1598662513453591583] adding template [.kibana-event-log-8.0.0-template] for index patterns [.kibana-event-log-8.0.0-*]
[00:00:00]                 │ info [o.e.c.m.MetadataCreateIndexService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1598662513453591583] [.kibana-event-log-8.0.0-000001] creating index, cause [api], templates [.kibana-event-log-8.0.0-template], shards [1]/[1]
[00:00:00]                 │ info [o.e.c.r.a.AllocationService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1598662513453591583] updating number_of_replicas to [0] for indices [.kibana-event-log-8.0.0-000001]
[00:00:00]                 │ info [o.e.x.i.IndexLifecycleTransition] [kibana-ci-immutable-ubuntu-16-tests-xxl-1598662513453591583] moving index [.kibana-event-log-8.0.0-000001] from [null] to [{"phase":"new","action":"complete","name":"complete"}] in policy [kibana-event-log-policy]
[00:00:00]                 │ info [o.e.c.r.a.AllocationService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1598662513453591583] current.health="GREEN" message="Cluster health status changed from [YELLOW] to [GREEN] (reason: [shards started [[.kibana-event-log-8.0.0-000001][0]]])." previous.health="YELLOW" reason="shards started [[.kibana-event-log-8.0.0-000001][0]]"
[00:00:00]                 │ info [o.e.x.i.IndexLifecycleTransition] [kibana-ci-immutable-ubuntu-16-tests-xxl-1598662513453591583] moving index [.kibana-event-log-8.0.0-000001] from [{"phase":"new","action":"complete","name":"complete"}] to [{"phase":"hot","action":"unfollow","name":"wait-for-indexing-complete"}] in policy [kibana-event-log-policy]
[00:00:00]                 │ info [o.e.x.i.IndexLifecycleTransition] [kibana-ci-immutable-ubuntu-16-tests-xxl-1598662513453591583] moving index [.kibana-event-log-8.0.0-000001] from [{"phase":"hot","action":"unfollow","name":"wait-for-indexing-complete"}] to [{"phase":"hot","action":"unfollow","name":"wait-for-follow-shard-tasks"}] in policy [kibana-event-log-policy]
[00:00:01]                 └- ✓ pass  (954ms) "apis SAML security SAML authentication does not prevent basic login"
[00:00:01]               └-: initiating handshake
[00:00:01]                 └-> "before all" hook
[00:00:03]               └-: finishing handshake
[00:00:03]                 └-> "before all" hook
[00:00:03]                 └-> should fail if SAML response is not complemented with handshake cookie
[00:00:03]                   └-> "before each" hook: global before each
[00:00:03]                   └-> "before each" hook
[00:00:03]                     │ proc [kibana]   log   [02:48:57.068] [info][plugins][routes][security] Logging in with provider "saml" (saml)
[00:00:04]                   │ info [o.e.x.s.a.AuthenticationService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1598662513453591583] Authentication to realm saml1 failed - Provided SAML response is not valid for realm saml/saml1 (Caused by org.elasticsearch.ElasticsearchSecurityException: SAML content is in-response-to [_7e7fa6a7bc4255833b2ecf409578c08362e4c8aa] but expected one of [])
[00:00:04]                   └- ✓ pass  (230ms) "apis SAML security SAML authentication finishing handshake should fail if SAML response is not complemented with handshake cookie"
[00:00:04]                 └-> should succeed if both SAML response and handshake cookie are provided
[00:00:04]                   └-> "before each" hook: global before each
[00:00:04]                   └-> "before each" hook
[00:00:04]                     │ proc [kibana]   log   [02:48:58.194] [info][plugins][routes][security] Logging in with provider "saml" (saml)
[00:00:04]                     │ info [o.e.c.m.MetadataCreateIndexService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1598662513453591583] [ilm-history-3-000001] creating index, cause [api], templates [ilm-history], shards [1]/[0]
[00:00:04]                     │ info [o.e.x.i.IndexLifecycleTransition] [kibana-ci-immutable-ubuntu-16-tests-xxl-1598662513453591583] moving index [ilm-history-3-000001] from [null] to [{"phase":"new","action":"complete","name":"complete"}] in policy [ilm-history-ilm-policy]
[00:00:04]                     │ info [o.e.c.r.a.AllocationService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1598662513453591583] current.health="GREEN" message="Cluster health status changed from [YELLOW] to [GREEN] (reason: [shards started [[ilm-history-3-000001][0]]])." previous.health="YELLOW" reason="shards started [[ilm-history-3-000001][0]]"
[00:00:04]                     │ info [o.e.x.i.IndexLifecycleTransition] [kibana-ci-immutable-ubuntu-16-tests-xxl-1598662513453591583] moving index [ilm-history-3-000001] from [{"phase":"new","action":"complete","name":"complete"}] to [{"phase":"hot","action":"unfollow","name":"wait-for-indexing-complete"}] in policy [ilm-history-ilm-policy]
[00:00:04]                     │ info [o.e.x.i.IndexLifecycleTransition] [kibana-ci-immutable-ubuntu-16-tests-xxl-1598662513453591583] moving index [ilm-history-3-000001] from [{"phase":"hot","action":"unfollow","name":"wait-for-indexing-complete"}] to [{"phase":"hot","action":"unfollow","name":"wait-for-follow-shard-tasks"}] in policy [ilm-history-ilm-policy]
[00:00:05]                   │ info [o.e.x.s.s.SecurityIndexManager] [kibana-ci-immutable-ubuntu-16-tests-xxl-1598662513453591583] security index does not exist. Creating [.security-tokens-7] with alias [.security-tokens]
[00:00:05]                   │ info [o.e.c.m.MetadataCreateIndexService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1598662513453591583] [.security-tokens-7] creating index, cause [api], templates [], shards [1]/[0]
[00:00:05]                   │ info [o.e.c.r.a.AllocationService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1598662513453591583] current.health="GREEN" message="Cluster health status changed from [YELLOW] to [GREEN] (reason: [shards started [[.security-tokens-7][0]]])." previous.health="YELLOW" reason="shards started [[.security-tokens-7][0]]"
[00:00:08]                   └- ✖ fail: apis SAML security SAML authentication finishing handshake should succeed if both SAML response and handshake cookie are provided
[00:00:08]                   │      Error: expected { username: 'a@b.c',
[00:00:08]                   │   roles: [],
[00:00:08]                   │   full_name: null,
[00:00:08]                   │   email: null,
[00:00:08]                   │   metadata: 
[00:00:08]                   │    { saml_nameid: 'a@b.c',
[00:00:08]                   │      'saml(urn:oid:0.0.7)': [ 'a@b.c' ],
[00:00:08]                   │      saml_nameid_format: 'urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified' },
[00:00:08]                   │   enabled: true,
[00:00:08]                   │   authentication_realm: { name: 'saml1', type: 'saml' },
[00:00:08]                   │   lookup_realm: { name: 'saml1', type: 'saml' },
[00:00:08]                   │   authentication_type: 'token',
[00:00:08]                   │   authentication_provider: 'saml' } to only have keys 'username', 'full_name', 'email', 'roles', 'metadata', 'enabled', 'authentication_realm', 'lookup_realm', and 'authentication_provider'
[00:00:08]                   │       at Assertion.assert (/dev/shm/workspace/parallel/7/kibana/packages/kbn-expect/expect.js:100:11)
[00:00:08]                   │       at Assertion.keys (/dev/shm/workspace/parallel/7/kibana/packages/kbn-expect/expect.js:505:8)
[00:00:08]                   │       at checkSessionCookie (test/saml_api_integration/apis/security/saml_login.ts:49:43)
[00:00:08]                   │ 
[00:00:08]                   │ 

```

#### Stack Trace
```
Error: expected { username: 'a@b.c',
  roles: [],
  full_name: null,
  email: null,
  metadata: 
   { saml_nameid: 'a@b.c',
     'saml(urn:oid:0.0.7)': [ 'a@b.c' ],
     saml_nameid_format: 'urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified' },
  enabled: true,
  authentication_realm: { name: 'saml1', type: 'saml' },
  lookup_realm: { name: 'saml1', type: 'saml' },
  authentication_type: 'token',
  authentication_provider: 'saml' } to only have keys 'username', 'full_name', 'email', 'roles', 'metadata', 'enabled', 'authentication_realm', 'lookup_realm', and 'authentication_provider'
    at Assertion.assert (/dev/shm/workspace/parallel/7/kibana/packages/kbn-expect/expect.js:100:11)
    at Assertion.keys (/dev/shm/workspace/parallel/7/kibana/packages/kbn-expect/expect.js:505:8)
    at checkSessionCookie (test/saml_api_integration/apis/security/saml_login.ts:49:43)
```
</details>

---

<details><summary>X-Pack SAML API Integration Tests.x-pack/test/saml_api_integration/apis/security/saml_login·ts.apis SAML security SAML authentication finishing handshake should succeed if both SAML response and handshake cookie are provided</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71136/testReport//junit/X-Pack%20SAML%20API%20Integration%20Tests/x-pack_test_saml_api_integration_apis_security_saml_login·ts/apis_SAML_security_SAML_authentication_finishing_handshake_should_succeed_if_both_SAML_response_and_handshake_cookie_are_provided_2)


#### Standard Out
```
Failed Tests Reporter:
  - Test has not failed recently on tracked branches

[00:00:00]       │
[00:00:00]         │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1598662513453591583] adding template [.kibana-event-log-8.0.0-template] for index patterns [.kibana-event-log-8.0.0-*]
[00:00:00]         └-: apis SAML
[00:00:00]           └-> "before all" hook
[00:00:00]           └-: security
[00:00:00]             └-> "before all" hook
[00:00:00]             └-: SAML authentication
[00:00:00]               └-> "before all" hook
[00:00:00]               └-> should reject API requests if client is not authenticated
[00:00:00]                 └-> "before each" hook: global before each
[00:00:00]                 │ info [o.e.c.m.MetadataCreateIndexService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1598662513453591583] [.kibana-event-log-8.0.0-000001] creating index, cause [api], templates [.kibana-event-log-8.0.0-template], shards [1]/[1]
[00:00:00]                 │ info [o.e.c.r.a.AllocationService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1598662513453591583] updating number_of_replicas to [0] for indices [.kibana-event-log-8.0.0-000001]
[00:00:00]                 └- ✓ pass  (72ms) "apis SAML security SAML authentication should reject API requests if client is not authenticated"
[00:00:00]               └-> does not prevent basic login
[00:00:00]                 └-> "before each" hook: global before each
[00:00:00]                 │ info [o.e.x.i.IndexLifecycleTransition] [kibana-ci-immutable-ubuntu-16-tests-xxl-1598662513453591583] moving index [.kibana-event-log-8.0.0-000001] from [null] to [{"phase":"new","action":"complete","name":"complete"}] in policy [kibana-event-log-policy]
[00:00:00]                 │ proc [kibana]   log   [02:08:01.876] [info][plugins][routes][security] Logging in with provider "basic" (basic)
[00:00:00]                 │ info [o.e.c.r.a.AllocationService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1598662513453591583] current.health="GREEN" message="Cluster health status changed from [YELLOW] to [GREEN] (reason: [shards started [[.kibana-event-log-8.0.0-000001][0]]])." previous.health="YELLOW" reason="shards started [[.kibana-event-log-8.0.0-000001][0]]"
[00:00:00]                 │ info [o.e.x.i.IndexLifecycleTransition] [kibana-ci-immutable-ubuntu-16-tests-xxl-1598662513453591583] moving index [.kibana-event-log-8.0.0-000001] from [{"phase":"new","action":"complete","name":"complete"}] to [{"phase":"hot","action":"unfollow","name":"wait-for-indexing-complete"}] in policy [kibana-event-log-policy]
[00:00:00]                 │ info [o.e.x.i.IndexLifecycleTransition] [kibana-ci-immutable-ubuntu-16-tests-xxl-1598662513453591583] moving index [.kibana-event-log-8.0.0-000001] from [{"phase":"hot","action":"unfollow","name":"wait-for-indexing-complete"}] to [{"phase":"hot","action":"unfollow","name":"wait-for-follow-shard-tasks"}] in policy [kibana-event-log-policy]
[00:00:01]                 └- ✓ pass  (954ms) "apis SAML security SAML authentication does not prevent basic login"
[00:00:01]               └-: initiating handshake
[00:00:01]                 └-> "before all" hook
[00:00:03]               └-: finishing handshake
[00:00:03]                 └-> "before all" hook
[00:00:03]                 └-> should fail if SAML response is not complemented with handshake cookie
[00:00:03]                   └-> "before each" hook: global before each
[00:00:03]                   └-> "before each" hook
[00:00:03]                     │ proc [kibana]   log   [02:08:04.929] [info][plugins][routes][security] Logging in with provider "saml" (saml)
[00:00:04]                   │ info [o.e.x.s.a.AuthenticationService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1598662513453591583] Authentication to realm saml1 failed - Provided SAML response is not valid for realm saml/saml1 (Caused by org.elasticsearch.ElasticsearchSecurityException: SAML content is in-response-to [_31ffd99ca822b07fbd99610242804493be2b9e39] but expected one of [])
[00:00:04]                   └- ✓ pass  (234ms) "apis SAML security SAML authentication finishing handshake should fail if SAML response is not complemented with handshake cookie"
[00:00:04]                 └-> should succeed if both SAML response and handshake cookie are provided
[00:00:04]                   └-> "before each" hook: global before each
[00:00:04]                   └-> "before each" hook
[00:00:04]                     │ proc [kibana]   log   [02:08:06.055] [info][plugins][routes][security] Logging in with provider "saml" (saml)
[00:00:05]                   │ info [o.e.x.s.s.SecurityIndexManager] [kibana-ci-immutable-ubuntu-16-tests-xxl-1598662513453591583] security index does not exist. Creating [.security-tokens-7] with alias [.security-tokens]
[00:00:05]                   │ info [o.e.c.m.MetadataCreateIndexService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1598662513453591583] [.security-tokens-7] creating index, cause [api], templates [], shards [1]/[0]
[00:00:05]                   │ info [o.e.c.r.a.AllocationService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1598662513453591583] current.health="GREEN" message="Cluster health status changed from [YELLOW] to [GREEN] (reason: [shards started [[.security-tokens-7][0]]])." previous.health="YELLOW" reason="shards started [[.security-tokens-7][0]]"
[00:00:08]                   └- ✖ fail: apis SAML security SAML authentication finishing handshake should succeed if both SAML response and handshake cookie are provided
[00:00:08]                   │      Error: expected { username: 'a@b.c',
[00:00:08]                   │   roles: [],
[00:00:08]                   │   full_name: null,
[00:00:08]                   │   email: null,
[00:00:08]                   │   metadata: 
[00:00:08]                   │    { saml_nameid: 'a@b.c',
[00:00:08]                   │      saml_nameid_format: 'urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified',
[00:00:08]                   │      'saml(urn:oid:0.0.7)': [ 'a@b.c' ] },
[00:00:08]                   │   enabled: true,
[00:00:08]                   │   authentication_realm: { name: 'saml1', type: 'saml' },
[00:00:08]                   │   lookup_realm: { name: 'saml1', type: 'saml' },
[00:00:08]                   │   authentication_type: 'token',
[00:00:08]                   │   authentication_provider: 'saml' } to only have keys 'username', 'full_name', 'email', 'roles', 'metadata', 'enabled', 'authentication_realm', 'lookup_realm', and 'authentication_provider'
[00:00:08]                   │       at Assertion.assert (/dev/shm/workspace/parallel/7/kibana/packages/kbn-expect/expect.js:100:11)
[00:00:08]                   │       at Assertion.keys (/dev/shm/workspace/parallel/7/kibana/packages/kbn-expect/expect.js:505:8)
[00:00:08]                   │       at checkSessionCookie (test/saml_api_integration/apis/security/saml_login.ts:49:43)
[00:00:08]                   │ 
[00:00:08]                   │ 

```

#### Stack Trace
```
Error: expected { username: 'a@b.c',
  roles: [],
  full_name: null,
  email: null,
  metadata: 
   { saml_nameid: 'a@b.c',
     saml_nameid_format: 'urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified',
     'saml(urn:oid:0.0.7)': [ 'a@b.c' ] },
  enabled: true,
  authentication_realm: { name: 'saml1', type: 'saml' },
  lookup_realm: { name: 'saml1', type: 'saml' },
  authentication_type: 'token',
  authentication_provider: 'saml' } to only have keys 'username', 'full_name', 'email', 'roles', 'metadata', 'enabled', 'authentication_realm', 'lookup_realm', and 'authentication_provider'
    at Assertion.assert (/dev/shm/workspace/parallel/7/kibana/packages/kbn-expect/expect.js:100:11)
    at Assertion.keys (/dev/shm/workspace/parallel/7/kibana/packages/kbn-expect/expect.js:505:8)
    at checkSessionCookie (test/saml_api_integration/apis/security/saml_login.ts:49:43)
```
</details>

---

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/65be40a3f04ef4348ba01d7015cb9da49838ea84) | diff | [baseline](https://github.com/elastic/kibana/commit/9873df8ee0ee88584db70debc3316126ba3205af) |
| --- | --- | --- | --- |
| ingestPipelines | 417 | +1 | 416 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/65be40a3f04ef4348ba01d7015cb9da49838ea84) | diff | [baseline](https://github.com/elastic/kibana/commit/9873df8ee0ee88584db70debc3316126ba3205af) |
| --- | --- | --- | --- |
| ingestPipelines | 715.6KB | +10.1KB | 705.5KB |

#### distributable file count
| id | [value](https://github.com/elastic/kibana/commit/65be40a3f04ef4348ba01d7015cb9da49838ea84) | diff | [baseline](https://github.com/elastic/kibana/commit/9873df8ee0ee88584db70debc3316126ba3205af) |
| --- | --- | --- | --- |
| total | 53207 | +1 | 53206 |

### History
* :broken_heart: [Build #70748](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70748/) failed 74ba900994fa8ca1190028212b183fdd9e488cfc

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70748/","number":"70748","commit":"74ba900994fa8ca1190028212b183fdd9e488cfc"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71136/","number":"71136","commit":"65be40a3f04ef4348ba01d7015cb9da49838ea84"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71425/)
* Commit: d55f580c6ab8c30e467984c08366f4251277817d
* This comment will update when the build is complete

### Failed CI Steps
* [X-Pack Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71425/execution/node/348/log)

### History
* :broken_heart: [Build #71136](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71136/) failed 65be40a3f04ef4348ba01d7015cb9da49838ea84
* :broken_heart: [Build #70748](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70748/) failed 74ba900994fa8ca1190028212b183fdd9e488cfc

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70748/","number":"70748","commit":"74ba900994fa8ca1190028212b183fdd9e488cfc"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71136/","number":"71136","commit":"65be40a3f04ef4348ba01d7015cb9da49838ea84"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71425/","number":"71425","commit":"d55f580c6ab8c30e467984c08366f4251277817d"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71780/)
* Commit: a6849406b5d5d83020563b9be29de684e88e8457
* This comment will update when the build is complete

### Failed CI Steps
* [Check i18n](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71780/execution/node/376/log)

### History
* :broken_heart: [Build #71425](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71425/) failed d55f580c6ab8c30e467984c08366f4251277817d
* :broken_heart: [Build #71136](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71136/) failed 65be40a3f04ef4348ba01d7015cb9da49838ea84
* :broken_heart: [Build #70748](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70748/) failed 74ba900994fa8ca1190028212b183fdd9e488cfc

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70748/","number":"70748","commit":"74ba900994fa8ca1190028212b183fdd9e488cfc"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71136/","number":"71136","commit":"65be40a3f04ef4348ba01d7015cb9da49838ea84"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71425/","number":"71425","commit":"d55f580c6ab8c30e467984c08366f4251277817d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71780/","number":"71780","commit":"a6849406b5d5d83020563b9be29de684e88e8457"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71877/)
* Commit: bad79134f5e1055e38a8df85e60e1aead1fcb5e9
* This comment will update when the build is complete

### Failed CI Steps
* [Check i18n](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71877/execution/node/394/log)

### History
* :broken_heart: [Build #71780](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71780/) failed a6849406b5d5d83020563b9be29de684e88e8457
* :broken_heart: [Build #71425](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71425/) failed d55f580c6ab8c30e467984c08366f4251277817d
* :broken_heart: [Build #71136](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71136/) failed 65be40a3f04ef4348ba01d7015cb9da49838ea84
* :broken_heart: [Build #70748](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70748/) failed 74ba900994fa8ca1190028212b183fdd9e488cfc

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70748/","number":"70748","commit":"74ba900994fa8ca1190028212b183fdd9e488cfc"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71136/","number":"71136","commit":"65be40a3f04ef4348ba01d7015cb9da49838ea84"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71425/","number":"71425","commit":"d55f580c6ab8c30e467984c08366f4251277817d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71780/","number":"71780","commit":"a6849406b5d5d83020563b9be29de684e88e8457"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71877/","number":"71877","commit":"bad79134f5e1055e38a8df85e60e1aead1fcb5e9"}]}
PIPELINE-->

mdefazio commented:
Here are some initial thoughts:
- I think we could make the Output flyout close when clicking outside of it.  
- Perhaps the 'Run the pipeline' button in the flyout could be above the code block/document block so it matches the placement of 'Refresh output' on the Output tab?
- Will we be able to provide the list of fields available in the processor configuration?
- Hitting 'Update' closes the flyout, so changing the configuration and viewing the output is difficult
- When updating the configuration, the 'Update' button should be disabled until a change is made.  Then perhaps we need to add a status icon next to the Output tab to indicate that the output is not current until 'Update' is clicked.  Closing the flyout might then also need an alert modal that asks to save/dismiss changes?
- Do we need a similar alert modal when cancelling the Edit Pipeline flow when there are edits and the user doesn't hit Save?

alisonelizabeth commented:
Thanks @mdefazio for the initial feedback!

> I think we could make the Output flyout close when clicking outside of it.

👍 I don't have a strong opinion about this, but don't mind implementing it.

> Perhaps the 'Run the pipeline' button in the flyout could be above the code block/document block so it matches the placement of 'Refresh output' on the Output tab?

For me, it feels a little cumbersome to first add JSON for the sample documents and then move your mouse back up to click "Run the pipeline". When filling out a form, I typically would expect the submit button to be below. I understand this is only one field though so we might treat it a little differently.

> Will we be able to provide the list of fields available in the processor configuration?

Can you clarify what you mean here? JL is working on building out the forms for each processor type.

> Hitting 'Update' closes the flyout, so changing the configuration and viewing the output is difficult
When updating the configuration, the 'Update' button should be disabled until a change is made. Then perhaps we need to add a status icon next to the Output tab to indicate that the output is not current until 'Update' is clicked. Closing the flyout might then also need an alert modal that asks to save/dismiss changes?
Do we need a similar alert modal when cancelling the Edit Pipeline flow when there are edits and the user doesn't hit Save?

If you click the "Output" tab, the simulation will rerun with your unsaved changes, so you are able to toggle back and forth between configuration and output and make changes. I understand this may not be obvious though. If you cancel, the output will be reset to the original state. I don't think we've typically disabled forms if the user has not changed any values. Similarly, we don't have confirmation modals for exiting forms with unsaved changes. I think the latter is needed, but just wanted to point out that it is probably an issue we need to solve across several of our apps with forms.

elasticmachine commented:
Pinging @elastic/es-ui (Team:Elasticsearch UI)

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72747/)
* Commit: 96161421c5a2390996043eff3e634392cf8c86d8

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/96161421c5a2390996043eff3e634392cf8c86d8) | diff | [baseline](https://github.com/elastic/kibana/commit/6e0b20cbe3a2a0645cec4421265e165b76afe2df) |
| --- | --- | --- | --- |
| ingestPipelines | 432 | +6 | 426 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/96161421c5a2390996043eff3e634392cf8c86d8) | diff | [baseline](https://github.com/elastic/kibana/commit/6e0b20cbe3a2a0645cec4421265e165b76afe2df) |
| --- | --- | --- | --- |
| ingestPipelines | 747.4KB | +15.9KB | 731.6KB |

### History
* :yellow_heart: [Build #71892](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71892/) was flaky d8107bc0871a5b99bb5761e6d904f8c09bdb4bbf
* :broken_heart: [Build #71877](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71877/) failed bad79134f5e1055e38a8df85e60e1aead1fcb5e9
* :broken_heart: [Build #71780](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71780/) failed a6849406b5d5d83020563b9be29de684e88e8457
* :broken_heart: [Build #71425](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71425/) failed d55f580c6ab8c30e467984c08366f4251277817d
* :broken_heart: [Build #71136](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71136/) failed 65be40a3f04ef4348ba01d7015cb9da49838ea84

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71136/","number":"71136","commit":"65be40a3f04ef4348ba01d7015cb9da49838ea84"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71425/","number":"71425","commit":"d55f580c6ab8c30e467984c08366f4251277817d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71780/","number":"71780","commit":"a6849406b5d5d83020563b9be29de684e88e8457"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71877/","number":"71877","commit":"bad79134f5e1055e38a8df85e60e1aead1fcb5e9"},{"status":"UNSTABLE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71892/","number":"71892","commit":"d8107bc0871a5b99bb5761e6d904f8c09bdb4bbf"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72747/","number":"72747","commit":"96161421c5a2390996043eff3e634392cf8c86d8"}]}
PIPELINE-->

jloleysens reviewed (approved):
Great work @alisonelizabeth !

I had one code change in view that would be great to get your thoughts on, but I view it as non-blocking for this contribution.

In terms of UX everything worked as expected from your PR description. I think it would be great if we can make the documents dropdown a little bit bigger:

<img width="303" alt="Screenshot 2020-09-07 at 14 57 23" src="https://user-images.githubusercontent.com/8155004/92390339-f7611580-f11a-11ea-99d8-52c9436329c5.png">

I really like the "Data in"/"Data out" sections added per processor. It gives users a lot of insight into the per-step transformations. I wonder if it is worth add a way to view these on the processor itself? I'd be interested to hear your thoughts @mdefazio !

jloleysens commented:
It is a little bit tricky to assess, but I am assuming we have not introduced changes that require test coverage changes.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74223/)
* Commit: 53f278bf41f85c4343d1a5d92ad5e2f461b60310

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/53f278bf41f85c4343d1a5d92ad5e2f461b60310) | diff | [baseline](https://github.com/elastic/kibana/commit/5c457d1e820320586db98ac965278d649ff7ef7c) |
| --- | --- | --- | --- |
| ingestPipelines | 437 | +6 | 431 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/53f278bf41f85c4343d1a5d92ad5e2f461b60310) | diff | [baseline](https://github.com/elastic/kibana/commit/5c457d1e820320586db98ac965278d649ff7ef7c) |
| --- | --- | --- | --- |
| ingestPipelines | 760.3KB | +15.9KB | 744.4KB |

### History
* :green_heart: [Build #72747](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72747/) succeeded 96161421c5a2390996043eff3e634392cf8c86d8
* :yellow_heart: [Build #71892](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71892/) was flaky d8107bc0871a5b99bb5761e6d904f8c09bdb4bbf
* :broken_heart: [Build #71877](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71877/) failed bad79134f5e1055e38a8df85e60e1aead1fcb5e9
* :broken_heart: [Build #71780](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71780/) failed a6849406b5d5d83020563b9be29de684e88e8457
* :broken_heart: [Build #71425](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71425/) failed d55f580c6ab8c30e467984c08366f4251277817d

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71425/","number":"71425","commit":"d55f580c6ab8c30e467984c08366f4251277817d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71780/","number":"71780","commit":"a6849406b5d5d83020563b9be29de684e88e8457"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71877/","number":"71877","commit":"bad79134f5e1055e38a8df85e60e1aead1fcb5e9"},{"status":"UNSTABLE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71892/","number":"71892","commit":"d8107bc0871a5b99bb5761e6d904f8c09bdb4bbf"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72747/","number":"72747","commit":"96161421c5a2390996043eff3e634392cf8c86d8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74223/","number":"74223","commit":"53f278bf41f85c4343d1a5d92ad5e2f461b60310"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74342/)
* Commit: 1aa0ef4c0131c6d5e65b28712d3ff45e292fdfc0

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/1aa0ef4c0131c6d5e65b28712d3ff45e292fdfc0) | diff | [baseline](https://github.com/elastic/kibana/commit/5c457d1e820320586db98ac965278d649ff7ef7c) |
| --- | --- | --- | --- |
| ingestPipelines | 447 | +16 | 431 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/1aa0ef4c0131c6d5e65b28712d3ff45e292fdfc0) | diff | [baseline](https://github.com/elastic/kibana/commit/5c457d1e820320586db98ac965278d649ff7ef7c) |
| --- | --- | --- | --- |
| ingestPipelines | 764.4KB | +20.0KB | 744.4KB |

### History
* :green_heart: [Build #74223](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74223/) succeeded 53f278bf41f85c4343d1a5d92ad5e2f461b60310
* :green_heart: [Build #72747](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72747/) succeeded 96161421c5a2390996043eff3e634392cf8c86d8
* :yellow_heart: [Build #71892](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71892/) was flaky d8107bc0871a5b99bb5761e6d904f8c09bdb4bbf
* :broken_heart: [Build #71877](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71877/) failed bad79134f5e1055e38a8df85e60e1aead1fcb5e9
* :broken_heart: [Build #71780](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71780/) failed a6849406b5d5d83020563b9be29de684e88e8457

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71780/","number":"71780","commit":"a6849406b5d5d83020563b9be29de684e88e8457"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71877/","number":"71877","commit":"bad79134f5e1055e38a8df85e60e1aead1fcb5e9"},{"status":"UNSTABLE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71892/","number":"71892","commit":"d8107bc0871a5b99bb5761e6d904f8c09bdb4bbf"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72747/","number":"72747","commit":"96161421c5a2390996043eff3e634392cf8c86d8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74223/","number":"74223","commit":"53f278bf41f85c4343d1a5d92ad5e2f461b60310"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74342/","number":"74342","commit":"1aa0ef4c0131c6d5e65b28712d3ff45e292fdfc0"}]}
PIPELINE-->

alisonelizabeth commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/context/processors_context.tsx:
```diff
@@ -149,6 +156,8 @@ export const PipelineProcessorsContextProvider: FunctionComponent<Props> = ({
           });
           break;
         case 'managingProcessor':
+          setUnsavedProcessorFormData(processorTypeAndOptions);
```
Great suggestion. I went ahead and implemented this, but only update the ref on submit rather than the `form.subscribe` callback, as the form data ends up getting reset on tab change here.

alisonelizabeth commented:
Thanks for the review @jloleysens! I addressed your feedback via https://github.com/elastic/kibana/pull/76058/commits/1aa0ef4c0131c6d5e65b28712d3ff45e292fdfc0. I also made the dropdown wider and fixed an alignment issue with the custom icons within the callout.

mdefazio reviewed (changes requested):
Some comments regarding design and UX:
- Let's change success `check` to a `checkInCircle` so it stands out more.  
- Change the dot icon color to `euiColorLightShade` so it _doesn't_ stand out more.  My thinking here is that by making more contrast between these it will be more obvious where in the list the pipeline stopped.
- Perhaps change 'Test' to 'Test pipeline' in the updated testing module (also would match the flyout header title) so it's clearer what this area does
- I think we could add a status and timestamp in the flyout that indicates that testing state.  I find myself unsure when it ran and what action in the UI will trigger a new run. (maybe a larger UI issue here?)
- Add a `play` icon to 'Run the pipeline' button in document tab since we have the refresh icon in the output tab
- 'Output' button in test module should read 'View output' so it represents an action
- Maybe not for this PR, but I noticed that the 'Refresh' button in the pipeline list is using the secondary color—this should be `primary` to match the Create button
![image](https://user-images.githubusercontent.com/3756330/93233425-0febb400-f749-11ea-91d3-df60657eaddc.png)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74541/)
* Commit: 7f87c1a013ca8ebc8348598ae7d49a7be569324b
* This comment will update when the build is complete

### Failed CI Steps
* [Jest Integration Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74541/execution/node/320/log)

### History
* :green_heart: [Build #74342](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74342/) succeeded 1aa0ef4c0131c6d5e65b28712d3ff45e292fdfc0
* :green_heart: [Build #74223](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74223/) succeeded 53f278bf41f85c4343d1a5d92ad5e2f461b60310
* :green_heart: [Build #72747](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72747/) succeeded 96161421c5a2390996043eff3e634392cf8c86d8
* :yellow_heart: [Build #71892](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71892/) was flaky d8107bc0871a5b99bb5761e6d904f8c09bdb4bbf
* :broken_heart: [Build #71877](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71877/) failed bad79134f5e1055e38a8df85e60e1aead1fcb5e9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71877/","number":"71877","commit":"bad79134f5e1055e38a8df85e60e1aead1fcb5e9"},{"status":"UNSTABLE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71892/","number":"71892","commit":"d8107bc0871a5b99bb5761e6d904f8c09bdb4bbf"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72747/","number":"72747","commit":"96161421c5a2390996043eff3e634392cf8c86d8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74223/","number":"74223","commit":"53f278bf41f85c4343d1a5d92ad5e2f461b60310"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74342/","number":"74342","commit":"1aa0ef4c0131c6d5e65b28712d3ff45e292fdfc0"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74541/","number":"74541","commit":"7f87c1a013ca8ebc8348598ae7d49a7be569324b"}]}
PIPELINE-->

alisonelizabeth commented:
@elasticmachine merge upstream

alisonelizabeth commented:
Thanks @mdefazio! Great suggestions.

I addressed most of your comments. Would you mind taking another look? I'm still not 100% on the best approach for adding a status/timestamp to the flyout. As mentioned 1:1, perhaps it's best to punt on this until we get some user testing.

> Maybe not for this PR, but I noticed that the 'Refresh' button in the pipeline list is using the secondary color—this should be primary to match the Create button

Great catch. I took a look at some of our other UIs, and we're also using the secondary color 🤔. The button was not introduced as part of this PR so I'm going to leave as-is for now, and it also might be something we need to address across other apps.

Latest screenshots:

<img width="1314" alt="Screen Shot 2020-09-15 at 12 35 26 PM" src="https://user-images.githubusercontent.com/5226211/93247270-0ae43000-f75c-11ea-9d3f-0d3ee9a6da27.png">

<img width="1246" alt="Screen Shot 2020-09-15 at 12 35 08 PM" src="https://user-images.githubusercontent.com/5226211/93247286-10417a80-f75c-11ea-8b70-86c792adb39a.png">

<img width="568" alt="Screen Shot 2020-09-15 at 12 09 21 PM" src="https://user-images.githubusercontent.com/5226211/93247303-1a637900-f75c-11ea-9fbe-1a11c58528f2.png">

mdefazio reviewed (approved):
Looks good.  I see we don't have the `euiColorLightShade` as an option for the icon.  Maybe this is something we can look into on the EUI side. But we'd obviously like to remove those hex values in a follow up PR.

kibanamachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74562/)
* Commit: b03f2d2c3dada9dc4ef26bd1ca8a044f2f63285b
* Flaky suites:
  * xpack-kibana-ciGroup7

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/b03f2d2c3dada9dc4ef26bd1ca8a044f2f63285b) | diff | [baseline](https://github.com/elastic/kibana/commit/9acf8d2656bacbe7a979dfe6d188813018c6205b) |
| --- | --- | --- | --- |
| ingestPipelines | 447 | +16 | 431 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/b03f2d2c3dada9dc4ef26bd1ca8a044f2f63285b) | diff | [baseline](https://github.com/elastic/kibana/commit/9acf8d2656bacbe7a979dfe6d188813018c6205b) |
| --- | --- | --- | --- |
| ingestPipelines | 764.4KB | +20.0KB | 744.4KB |

### History
* :broken_heart: [Build #74541](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74541/) failed 7f87c1a013ca8ebc8348598ae7d49a7be569324b
* :green_heart: [Build #74342](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74342/) succeeded 1aa0ef4c0131c6d5e65b28712d3ff45e292fdfc0
* :green_heart: [Build #74223](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74223/) succeeded 53f278bf41f85c4343d1a5d92ad5e2f461b60310
* :green_heart: [Build #72747](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72747/) succeeded 96161421c5a2390996043eff3e634392cf8c86d8
* :yellow_heart: [Build #71892](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71892/) was flaky d8107bc0871a5b99bb5761e6d904f8c09bdb4bbf

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"UNSTABLE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71892/","number":"71892","commit":"d8107bc0871a5b99bb5761e6d904f8c09bdb4bbf"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72747/","number":"72747","commit":"96161421c5a2390996043eff3e634392cf8c86d8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74223/","number":"74223","commit":"53f278bf41f85c4343d1a5d92ad5e2f461b60310"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74342/","number":"74342","commit":"1aa0ef4c0131c6d5e65b28712d3ff45e292fdfc0"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74541/","number":"74541","commit":"7f87c1a013ca8ebc8348598ae7d49a7be569324b"},{"status":"UNSTABLE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74562/","number":"74562","commit":"b03f2d2c3dada9dc4ef26bd1ca8a044f2f63285b"}]}
PIPELINE-->
